src/app/modules/main/wallet_section/activity/controller.nim

```diff
@@ -0,0 +1,131 @@
+      let response = backend.getTransfersForIdentities(transactionIdentities)
+      let res = response.result
+      if response.error != nil or res.kind != JArray or res.len == 0:
+        raise newException(Defect, "failed fetching transaction details")
+
+      let transactionsDtos = res.getElems().map(x => x.toTransactionDto())
+      transactions = self.transactionsModule.transactionsToItems(transactionsDtos, @[])
+
+    var pendingTransactions: seq[Item] = @[]
+    if len(pendingTransactionIdentities) > 0:
+      let response = backend.getPendingTransactionsForIdentities(pendingTransactionIdentities)
+      let res = response.result
+      if response.error != nil or res.kind != JArray or res.len == 0:
+        raise newException(Defect, "failed fetching pending transactions details")
+
+      let pendingTransactionsDtos = res.getElems().map(x => x.toPendingTransactionDto())
+      pendingTransactions = self.transactionsModule.transactionsToItems(pendingTransactionsDtos, @[])
+
+    # Merge detailed transaction info in order
+    result = newSeq[entry.ActivityEntry](multiTransactions.len + transactions.len + pendingTransactions.len)
+    var mtIndex = 0
+    var tIndex = 0
+    var ptIndex = 0
+    for i in low(backendEnties) .. high(backendEnties):
+      let backendEntry = backendEnties[i]
+      case backendEntry.transactionType:
+        of MultiTransaction:
+          result[i] = entry.newMultiTransactionActivityEntry(multiTransactions[mtIndex])
+          mtIndex += 1
+        of SimpleTransaction:
+          let refInstance = new(Item)
+          refInstance[] = transactions[tIndex]
+          result[i] = entry.newTransactionActivityEntry(refInstance, false)
+          tIndex += 1
+        of PendingTransaction:
+          let refInstance = new(Item)
+          refInstance[] = pendingTransactions[ptIndex]
+          result[i] = entry.newTransactionActivityEntry(refInstance, true)
+          ptIndex += 1
+
+  proc refreshData*(self: Controller) {.slot.} =
+    # result type is RpcResponse
+    let response = backend_activity.getActivityEntries(@["0x0000000000000000000000000000000000000001"], @[1], self.currentActivityFilter, 0, 10)
+    # RPC returns null for result in case of empty array
+    if response.error != nil or (response.result.kind != JArray and response.result.kind != JNull):
+      error "error fetching activity entries: ", response.error
+      return
+
+    if response.result.kind == JNull:
+      self.model.setEntries(@[])
+      return
+
+    var backendEnties = newSeq[backend_activity.ActivityEntry](response.result.len)
+    for i in 0 ..< response.result.len:
+      backendEnties[i] = fromJson(response.result[i], backend_activity.ActivityEntry)
+    let entries = self.backendToPresentation(backendEnties)
+    self.model.setEntries(entries)
+
+  # TODO: add all parameters and separate in different methods
+  proc updateFilter*(self: Controller, startTimestamp: int, endTimestamp: int) {.slot.} =
```

Comment:
This is just a debugging endpoint. In the follow-up PR will add more, however, I think each filter criteria should have its own slot and update the self.currentActivityFilter then call update.

Reply:
We should probably have some common terminology here. So, the filtering criteria we end up using to fetch stuff from status-go should be the composition of two parts:
The "content-independent" (what Sale calls top-down, I believe) criteria: Addresses, chains. This should affect Assets, Collectibles and Activity at the same time. What we select for this should affect which criteria becomes available for the other part.
The "content-dependent" (bottom-up) criteria: Depends on what you're trying to get in response / the specific tab you're in. We currently only have this for activity (but you could for example filter collectibles by collection, date of transfer, date of creation, etc. Basically whatever OpenSea offers on their website), and it will be all the stuff listed in the Figma designs (period, type, status, token symbol/NFT name, counterparty)
AFAI, we're only dealing with the second part here.

Reply:
> I think each filter criteria should have its own slot and update the self.currentActivityFilter then call update

This is the thing we discussed in one of our brainstorming meetings, but seems we're still not on the same page. 

> We should probably have some common terminology here

A simple small diagram with relations what is owned by what and data/signal flow directions would be perfect, as well for us during the development and also for any new dev.